src/main/java/org/jenkins/tools/test/hook/Jetty12Hook.java

```diff
@@ -0,0 +1,66 @@
+package org.jenkins.tools.test.hook;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.util.VersionNumber;
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.util.jar.JarFile;
+import java.util.jar.JarInputStream;
+import java.util.jar.Manifest;
+import java.util.zip.ZipEntry;
+import org.jenkins.tools.test.model.hook.BeforeExecutionContext;
+import org.jenkins.tools.test.model.hook.PluginCompatTesterHookBeforeExecution;
+import org.kohsuke.MetaInfServices;
+
+/**
+ * Ensure that if the core is on Jetty 12, that the test harness is on Jetty 12 as well.
+ */
+@MetaInfServices(PluginCompatTesterHookBeforeExecution.class)
+public class Jetty12Hook extends PropertyVersionHook {
+
+    @Override
+    public String getProperty() {
+        return "jenkins-test-harness.version";
+    }
+
+    @Override
+    public String getMinimumVersion() {
+        return "2230.v4fa_477b_634f4";
+    }
+
+    @Override
+    public boolean check(@NonNull BeforeExecutionContext context) {
+        VersionNumber winstoneVersion = getWinstoneVersion(context.getConfig().getWar());
+        if (winstoneVersion.getDigitAt(0) < 7) {
+            return false;
+        }
+        return super.check(context);
```

Comment:
… but otherwise invoke the logic in the superclass, which dynamically updates `jenkins-test-harness.version` to `2230.v4fa_477b_634f4`, but only if the current value of the version is older (newer values don't need to be overridden).